kotlin/src/test/kotlin/com/linecorp/armeria/server/FlowAnnotatedServiceTest.kt

```diff
@@ -0,0 +1,278 @@
+            "\u001E{\"name\":\"foo\",\"age\":10}\n",
+            "\u001E{\"name\":\"bar\",\"age\":20}\n",
+            "\u001E{\"name\":\"baz\",\"age\":30}\n",
+            ""
+        )
+    }
+
+    @Test
+    fun test_eventStreaming() = runBlocking {
+        client.get("/flow/event-streaming") shouldProduce listOf(
+            "id:1\n" + "event:MESSAGE_DELIVERED\n" + "data:{\"message_id\":1}\n" + "\n",
+            "id:2\n" + "event:FOLLOW_REQUEST\n" + "data:{\"user_id\":123}\n" + "\n",
+            ""
+        )
+    }
+
+    @Test
+    @Disabled
+    fun test_blockingAnnotation(): Unit = runBlocking {
+        val res = client.get("/flow/blocking-annotation").aggregate().await()
+        assertThat(res.status()).isEqualTo(HttpStatus.OK)
+        assertThat(res.contentUtf8()).isEqualTo("OK")
+    }
+
+    @Test
+    fun test_customContext(): Unit = runBlocking {
+        val res = client.get("/flow/custom-context").aggregate().await()
+        assertThat(res.status()).isEqualTo(HttpStatus.OK)
+        assertThat(res.contentUtf8()).isEqualTo("OK")
+    }
+
+    @Test
+    fun test_customDispatcher(): Unit = runBlocking {
+        val res = client.get("/flow/custom-dispatcher").aggregate().await()
+        assertThat(res.status()).isEqualTo(HttpStatus.OK)
+        assertThat(res.contentUtf8()).isEqualTo("OK")
+    }
+
+    @Test
+    fun test_cancellation() {
+        try {
+            client.get("/flow/cancellation").aggregate().get()
+        } catch (e: Exception) {
+            // do nothing
+        }
+        Thread.sleep(3000L)
+        assertThat(cancelled.get()).isTrue
+    }
+
+    companion object {
+        @JvmField
+        @RegisterExtension
+        val server = object : ServerExtension() {
+            override fun configure(sb: ServerBuilder) {
+                @Suppress("unused")
+                sb.apply {
+                    annotatedService("/flow", object {
+                        @Get("/byte-streaming")
+                        @ProducesOctetStream
+                        suspend fun byteStreaming(): Flow<ByteArray> = flow {
+                            emit("hello".toByteArray())
+                            emit("world".toByteArray())
+                        }
+
+                        @Get("/json-string-streaming")
+                        @ProducesJsonSequences
+                        suspend fun jsonStreamingString(): Flow<String> = flow {
+                            emit("hello")
+                            emit("world")
+                        }
+
+                        @Get("/json-obj-streaming")
+                        @ProducesJsonSequences
+                        suspend fun jsonStreamingObj(): Flow<Member> = flow {
+                            emit(Member(name = "foo", age = 10))
+                            emit(Member(name = "bar", age = 20))
+                            emit(Member(name = "baz", age = 30))
+                        }
```

Comment:
Could you add a test for `@ProducesJson`?

Reply:
Added!